minecraft/1.7/src/main/java/nova/core/wrapper/mc/forge/v17/wrapper/block/backward/BWBlock.java

```diff
@@ -69,9 +69,10 @@ public BWBlock(net.minecraft.block.Block block) {
 	public BWBlock(net.minecraft.block.Block block, World world, Vector3D pos) {
 		this.mcBlock = block;
 		components.add(new BWBlockTransform(this, world, pos));
-		components.add(new BlockProperty.Opacity().setOpacity(mcBlock.getMaterial().isOpaque() ? 1 : 0));
-		if (mcBlock.isReplaceable(blockAccess(), xi(), yi(), zi()))
-			components.add(BlockProperty.Replaceable.instance());
+		components.add(new BlockProperty.Opacity()).setOpacity(() -> mcBlock.getMaterial().isOpaque() ? 1 : 0);
+		Optional.of(components.add(new BlockProperty.Replaceable()))
+			.filter(r -> block != Blocks.air)
```

Comment:
Isn't this just overuse of `Optional` when you can just use `if (block != Blocks.air)` lol? The key here is that there's no `null` being dealt with so `Optional` is wrong. Correct me if i'm wrong ofc.

Reply:
I wanted to avoid an instance variable.